src/pages/Integrations/List/List.tsx

```diff
@@ -225,6 +226,50 @@ const IntegrationsList: React.FunctionComponent<IntegrationListProps> = ({
       integrations.count - (pageData.page.index - 1) * pageData.page.size
     ) || 10;
 
+  const integrationsModals: JSX.Element[] = [];
+
+  if (modalIsOpenState.isOpen && !wizardEnabled) {
+    integrationsModals.push(
+      <CreatePage
+        isEdit={modalIsOpenState.isEdit}
+        initialIntegration={modalIsOpenState.template || {}}
+        onClose={closeFormModal}
+      />
+    );
+  }
+
+  if (wizardEnabled) {
+    integrationsModals.push(
+      <CreateWizard
+        isOpen={modalIsOpenState.isOpen}
+        closeModal={() => modalIsOpenActions.reset()}
+        category={category}
+      />
+    );
+  }
+
+  if (deleteModalState.data) {
+    integrationsModals.push(
+      <IntegrationDeleteModalPage
+        onClose={closeDeleteModal}
+        integration={deleteModalState.data}
+      />
+    );
+  }
```

Comment:
@karelhala I'm not a big fan of conditional rendering in the JSX itself - especially as the JSX grows - it becomes much harder to debug what is rendered and why. It gets even worse as we add to the conditionals, and can result in unexpected things rendering as the conditions may be linked together. Whereas returning fast/immediately seems much clearer to me (paying close attention to avoid DOM duplication). 

IE (pseudo code, but like I had it):
```tsx
if (empty) {
  return <EmptyState .../>
}

return <IntegrationsUI .../>
```
This seems cleaner to me - and avoids the risk of accidentally rendering the IntegrationsUI and the EmptyState at the same time.

versus:
```tsx
return (
  <>
    {empty && (<EmptyState ... />)}
    {!empty && (<IntegrationsUI .../>)}
  </>
);
```

However I see we are doing that already in the codebase - and I'm not yet willing to lie down on the tracks for it. So for now I've updated to as you requested I think. 

Regarding your point of having the `integrationsModals` in two places being confusing - that is not strictly required. I did it that way to keep the PR size small as refactoring the modal setup logic and abstracting it outside of `List.tsx` is another effort in and of itself. I do think we should consider this blog post around how our components should be implemented: https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0 (sorry if I'm a broken record).

**TL;DR** - updated as you requested for now; we should revisit this style as a broader discussion.

Reply:
Sure thing! I completely understand you, let's bring this up with the wider team and discuss what's easier for us to read and mainaint.